dymos/phase/phase.py

```diff
@@ -1083,9 +1083,9 @@ def add_boundary_constraint(self, name, loc, constraint_name=None, units=None,
         loc : str
```

Comment:
The docstring description for "name" should also be modified to explain that it can be an expression of variables.